src/utils/getCustomAPIWalletAllowed.ts

```diff
@@ -0,0 +1,23 @@
+import axios from "axios";
+import { CustomAPIWalletAllowed } from "../types";
+
+export const getCustomAPIWalletAllowed = async (
+    apiUrl: string,
+    walletAddress: string,
+): Promise<boolean> => {
+  let walletAllowedRes;
+  const finalApiUrl: string = apiUrl.replace("$(wallet)", walletAddress);
+  
+  try {
+    walletAllowedRes = (
+      await axios.get(finalApiUrl)
+    ).data as CustomAPIWalletAllowed;
+
+    return walletAllowedRes.allowed;
+
+  } catch (e) {
+    console.log(e);
+
+    throw new Error("Failed to request the custom api");
```

Comment:
Instead of throwing an error here, let's return false.

When an error is returned, it stops the execution of `coldUpdateDiscordRoles` which means roles will not be completely updated.

Ideally:
- When random earth throws an error, don't update any roles because we rely on random earth for data across all discord servers, so throw an error.
- When knowhere throwns an error, don't update any roles because we rely on knowhere for data across all discord servers, so throw an error.
- When an api rule throws an error, imagine it is false. Don't want that to block the rest of the updates if one server has a bad api. Instead punish that specific server by removing roles.
- When querying a CW20 or CW721 contract throws an error, imagine it is false. This is problematic if the error was due to rate limiting because we may remove roles from deserving people, but the alternative of letting people keep roles that they don't deserve is just as bad. The solution should rate limiting errors appear is to find a way to optimize and get rid of the rate limiting errors.

We are almost at this ideal, except need to update `getCustomAPIWalletAllowed` to return false when there is an error.